### PR TITLE
Clamp blood loss to range [0...100]

### PR DESCRIPTION
**Pull request by OrenAudeles**
PR #23375 Text -- 
> Evidently if you debug your current hp above your max hp body temp goes nuts.
> Not an actual bug, but still reasonable to clamp the offending value to a reasonable (non-negative in this case)

Clamps blood ratio to [0...100] so that if the player debugs their bodypart HP above their max bodypart HP it doesn't cause overheating/freezing.

### CODE REVIEW

**Comment by AMurkin** [comment]
```cpp
    } else if( bp == bp_torso ) {
        hp_cur_sum = hp_cur[hp_torso];
        hp_max_sum = hp_max[hp_torso];
    } else if( bp == bp_head ) {
        hp_cur_sum = hp_cur[hp_head];
        hp_max_sum = hp_max[hp_head];
    }
```
Can be written as
```cpp
    } else if( bp == bp_torso || bp == bp_head ) {
        hp_cur_sum = hp_cur[bp];
        hp_max_sum = hp_max[bp];
    }
```

File: src/player.cpp [hunk 1]
```diff
@@ -1667,21 +1667,28 @@ int player::temp_corrected_by_climate_control( int temperature ) const
...
+        hp_cur_sum = hp_cur[hp_torso];
+        hp_max_sum = hp_max[hp_torso];
     } else if( bp == bp_head ) {
-        blood_loss = ( 100 - 100 * hp_cur[hp_head] / hp_max[hp_head] );
+        hp_cur_sum = hp_cur[hp_head];
+        hp_max_sum = hp_max[hp_head];
     }
-    return blood_loss;
+    
+    hp_cur_sum = std::min(hp_max_sum, std::max(0, hp_cur_sum));
```
Comment (AMurkin) [inline src/player.cpp]: Spaces in (). And in the next line.

Comment (AMurkin) [reply src/player.cpp]: Yes.
